### DISCUSSION TITLE
PyAEDT Community Rules

### DISCUSSION THREAD

**Discussion by Samuelopez-ansys** [ansys/pyaedt#3809] [Announcements]
Dear **Contributors** and **PyAEDT lovers**,

To ensure productive and respectful discussions, we've implemented the following rules:

- Be respectful and considerate: We're in this project together, so let's treat everyone with respect.
- Stay on topic: Please keep the discussion relevant to the issue at hand.
- Use clear and concise language: To ensure effective communication, please keep your language clear.

**Directing Issues and Features Requests**

For reporting bugs and proposing new features, please use the `Issues` tab instead of the `Discussions` tab. This will help us track and prioritize work in a more organized manner.

**Purpose of Discussions**

The `Discussions` tab should primarily be used for general questions and discussions about the project. This could include discussions about potential improvements, the future direction of the project, brainstorming ideas, help with using the software, and other topics that don't exactly fit as an `Issue`.

Remember, maintaining distinct places for different types of interactions helps keep our project organized and moving forward in a coordinated manner.

**Everyone is Welcome**

Everyone can contribute to this project, irrespective of their level of expertise. Your diverse skills, perspectives, and experiences are valuable and we welcome them. 

Here's a simple overview of how you can start making contributions:

1. **Fork the Repository**: "Forking" means creating a personal copy of this repository on your GitHub account.
   
2. **Clone the Repository**: After forking, you need to download the repository to your local machine. This can be done using the `git clone` command.

3. **Create a New Branch**: A branch is used to isolate your changes from the main project. You can create a new branch using the `git branch` command. Remember to switch to your new branch with the `git checkout` command.

4. **Commit Your Changes**: After making your changes, you need to "commit" them. A commit is a packaged set of alterations. Use `git add` to add your files to staging, and then `git commit -m "your message"` to commit them.

5. **Push Your Changes**: After committing your changes, "push" them to your forked repository on GitHub with `git push origin <branch-name>`.

6. **Create a Pull Request**: A Pull Request (PR) lets us know you have changes you think should be included in the main project. Go to your forked repository on GitHub and click on the 'Pull request' button.

Following these steps ensures that your contributions will be easily reviewed and potentially included in the project much faster.

Please don't get discouraged if you find these steps complex, we are here to help you throughout the process.

We hope these rules will make the Discussions section a better place for every contributor.

Thank you!

**Comment by ghost** [comment]
at what point are the automated tests run in the steps above? Is it possible to run the tests on code that is not ready to be pushed?

**Reply by Samuelopez-ansys** [reply]
@pyAnsysUser You can run the tests locally. First install the test dependencies: pip install .[tests] 

Then you just need to run any of the tests inside the _unittest folder.

It is using pytest, you can also run the tests in the terminal:

pytest test_mod.py

**Comment by raghav-pyaedt** [comment]
Hello, 
I have AEDT 25R2 installed in my machine. I want to install PYAEDT and extension tools like Via wizard. I'm facing issues. Any help to get me pyaedt installed into AEDT would be greatly appreciated!

1)
I was trying to install the pyAEDT from this location below but I get the error when I try to run the script in AEDT tools > run script
[Installation — PyAEDT](https://aedt.docs.pyansys.com/version/stable/Getting_started/Installation.html#)
 
the log file says 
OSError: [WinError 4551] Your organization used Device Guard to block this app. Contact your support person for more info
 
2)
I have Python 3.13.3 installed as well as Python PIP installed. 

I have used wheelhouse 3.10 to install along with pyaedt_installer_from_aedt but it gives out error 
"
Traceback (most recent call last):
  File "C:\Users\*Username*\Downloads\pyaedt_installer_from_aedt.py", line 537, in <module>
    install_pyaedt()
  File "C:\Users\*Username*\Downloads\pyaedt_installer_from_aedt.py", line 454, in install_pyaedt
    subprocess.run(command, check=True, env=env)  # nosec
  File "C:\Program Files\ANSYS Inc\v252\AnsysEM\commonfiles\CPython\3_10\winx64\Release\python\lib\subprocess.py", line 503, in run
    with Popen(*popenargs, **kwargs) as process:
  File "C:\Program Files\ANSYS Inc\v252\AnsysEM\commonfiles\CPython\3_10\winx64\Release\python\lib\subprocess.py", line 971, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Program Files\ANSYS Inc\v252\AnsysEM\commonfiles\CPython\3_10\winx64\Release\python\lib\subprocess.py", line 1456, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] The system cannot find the file specified"
